homeassistant/config_entries.py

```diff
@@ -265,9 +266,44 @@ class ConfigFlowResult(FlowResult, total=False):
 
     minor_version: int
     options: Mapping[str, Any]
+    subentries: Iterable[ConfigSubentryData]
     version: int
 
 
+class ConfigSubentryData(TypedDict):
+    """Container for configuration subentry data."""
+
+    data: Mapping[str, Any]
+    subentry_id: str
+    title: str
+    unique_id: str | None
+
+
+class SubentryFlowResult(FlowResult, total=False):
+    """Typed result dict for subentry flow."""
+
+    unique_id: str | None
+
+
+@dataclass(frozen=True, kw_only=True)
+class ConfigSubentry:
+    """Container for a configuration subentry."""
+
+    data: MappingProxyType[str, Any]
+    subentry_id: str = field(default_factory=uuid_util.random_uuid_hex)
```

Comment:
```suggestion
    subentry_id: str = field(default_factory=uuid_util.random_uuid_hex)
```
Should this use `ulid` as in ConfigEntires and others?

Reply:
I'm using the same random id generator as for config entries: https://github.com/home-assistant/core/blob/792d1ad2cf087127799fede6d9db004eaaaf2b55/homeassistant/config_entries.py#L365

I'm not sure why we don't use ulid for new config entries, maybe it's simply because config entries are not created very often.